cmdx.py

```diff
@@ -2627,7 +2674,27 @@ def __str__(self):
         return str(self.read())
 
     def __repr__(self):
-        return str(self.read())
+        try:
+            # Delegate the value reading to __str__
+            read_result = str(self)
+            valid = True
+        except:
+            valid = False
+
+        cls_name = '{}.{}'.format(__name__, self.__class__.__name__)
+        msg = '{}("{}", "{}")'.format(cls_name,
+                                      self.node().name(),
+                                      self.name())
+        if valid:
+            try:
+                if self.typeClass() == String:
```

Comment:
Since we've established that typeClass isn't relevant for all types, like kByte, how about we use Maya's apiType here instead? Especially since we're only dealing with a single type.

- https://github.com/mottosso/cmdx/blob/344aed050aa8dec79566d479de354f4a94ff0d4b/cmdx.py#L3868

It would save on the use of try/except too.

Did we also have a test for this case?

Reply:
> If this is not the case anymore, and we never expect str to fail, would it be ok to expect that it can return None? In the case of a Compound attribute for instance? That way I can add that check to include == or not.

Sure, let's try it.

> A third option, albeit more complex, could be to have it return a dictionary, with the child attribute names as keys for instance.

I mean, this would be supremely useful. But I'm uncertain how stable we can make this.. Maybe let's start small, and add this later on?

Reply:
> Sure, let's try it

Scratch that. We can't return something that isn't a string type from `__str__`. That raises a `TypeError`, silly me.

I think I was able to think of a decent solution within `__repr__`, so we're not messing with `__str__` in this PR. pushing a commit now which includes this and the [above](https://github.com/mottosso/cmdx/pull/78#discussion_r1500286153)